### ISSUE TITLE
RGB to grayscale conversion can fail during inference

### ISSUE THREAD

**Issue by talmo** [murthylab/sleap#638]
SLEAP auto-detects whether videos are RGB or grayscale and loads them as grayscale prior to any neural network operation. This might fail, however, if the color channels are slightly different, e.g., due to codec or compression settings.

Models that expect grayscale inputs should automatically detect RGB image input and handle it appropriately regardless, however this fails on the GPU (but not on the CPU!) for some reason, resulting in errors like this:

```
InternalError: 2 root error(s) found.
  (0) INTERNAL:  Blas xGEMV launch failed : a.shape=[1,1310720,3], b.shape=[1,3,1], m=1310720, n=1, k=3
	 [[node top_down_inference_model/centroid_crop/rgb_to_grayscale/Tensordot/MatMul
 (defined at d:\sleap_develop\sleap\nn\data\normalization.py:94)
]]
	 [[top_down_inference_model/find_instance_peaks/range/_204]]
  (1) INTERNAL:  Blas xGEMV launch failed : a.shape=[1,1310720,3], b.shape=[1,3,1], m=1310720, n=1, k=3
	 [[node top_down_inference_model/centroid_crop/rgb_to_grayscale/Tensordot/MatMul
 (defined at d:\sleap_develop\sleap\nn\data\normalization.py:94)
]]
0 successful operations.
0 derived errors ignored. [Op:__inference_predict_function_7183]

Errors may have originated from an input operation.
Input Source operations connected to node top_down_inference_model/centroid_crop/rgb_to_grayscale/Tensordot/MatMul:
In[0] top_down_inference_model/centroid_crop/rgb_to_grayscale/Tensordot/Reshape:	
In[1] top_down_inference_model/centroid_crop/rgb_to_grayscale/Tensordot/Reshape_1:

...

>>>   File "d:\sleap_develop\sleap\nn\inference.py", line 1368, in call
>>>     imgs = self.preprocess(imgs)
>>> 
>>>   File "d:\sleap_develop\sleap\nn\inference.py", line 742, in preprocess
>>>     if self.ensure_grayscale:
>>> 
>>>   File "d:\sleap_develop\sleap\nn\inference.py", line 743, in preprocess
>>>     imgs = sleap.nn.data.normalization.ensure_grayscale(imgs)
>>> 
>>>   File "d:\sleap_develop\sleap\nn\data\normalization.py", line 93, in ensure_grayscale
>>>     if image.shape[-1] == 3:
>>> 
>>>   File "d:\sleap_develop\sleap\nn\data\normalization.py", line 94, in ensure_grayscale
>>>     return tf.image.rgb_to_grayscale(image)
>>> 

Function call stack:
predict_function -> predict_function
```

This can be mitigated by adjusting for grayscale at the data pipeline stage, which can be detected from the model input layer.

**Comment by outlace** [comment]
Yes I just got this error. What do you mean adjust grayscale at the data pipeline stage?

**Comment by talmo** [comment]
Hi @outlace,

That was mostly to document what needed to change in the internal functionality. The fix will be included in the next SLEAP (pre-)release -- no need for you to do anything.

If you don't want to wait, you can try the latest version now like this:
```
git clone https://github.com/murthylab/sleap sleap_develop
cd sleap_develop
conda env create -f environment.yml -n sleap_develop
```

Then just use SLEAP as you normally would, except you'll want to `conda activate sleap_develop`  to use the latest version.

Cheers,

Talmo

**Comment by talmo** [comment]
This is now fixed in v1.2.0.
